agora/Agora/Stake/Redeemers.hs

```diff
@@ -293,31 +292,33 @@ premoveLocks =
             ( (pfield @"createdAt" #) ->
                 createdAt
               ) =
-              let notInCooldown =
-                    pmaybe
-                      # pconstant True
-                      # plam (\c -> createdAt + c #<= lowerBound)
-                      # unlockCooldown
+              let notInCooldown = createdAt + unlockCooldown #<= lowerBound
                in foldl1
                     (#||)
                     [ shouldRemoveAllLocks
-                    , ptraceIfFalse "Stake lock in cooldown" notInCooldown
+                    , -- Fail the transaction if a voter lock is in cooldown.∏
+                      passert
```

Comment:
I find this hard to digest. On one hand there's an or `#||`, on the other hand we only accept the right side to be true. So the value of `shouldRemoveAllLocks` doesn't matter, right?

Reply:
Well, if `shouldRemoveAllLocks` is true the `passert` part won't be evaluated.

Reply:
Then I think an `if` statement would be easier to read. Something like `pif shouldRemoveAllLocks (pconstant True) (passert ...)`.